src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -78,52 +80,51 @@ public void showAllTaskEvents(){
 					itemToAdd = nextDeadlineTask;
 					deadlineTaskCounter++;
 				}
-		    }
-		    
-		    LocalDate tempDate = getLatestDate(currentDate, itemToAdd);
-		    
-		    if(currentDate == null){
-		    	currentDate = tempDate;
-		    	addLabel(currentDate);
-		    }
-		    else if(!currentDate.equals(tempDate)){
-		    	currentDate = tempDate;
-		    	addLabel(currentDate);
-		    }
-		    
-		    addTaskEvent(itemToAdd);
+			}
+
+			LocalDate tempDate = getLatestDate(currentDate, itemToAdd);
```

Comment:
Why do you need to get the later between currentDate and nextDate? To avoid repeating all the null checking stuff, you can just do initialize nextDate in the if else conditions above. For currentDate just initialize it to `LocalDate.MIN` so that you don't have to check whether it's null.

After you've had `currentDate` and `nextDate` just check whether they are the same to add the day header.